cmd/server/main.go

```diff
@@ -239,3 +235,63 @@ func downloadPluginFile(name, version string) (*os.File, int64, error) {
 
 	return tmpfile, written, nil
 }
+
+func lookupOrScanPlugin(plugin string, version string) (*shared.Scan, error) {
```

Comment:
Currently, doing a diff against an unknown version will do the download/scan/store procedure because that is fully locked up in this function. I think it makes sense to memoize the results, but I do wonder if that's the best idea.